### PR TITLE
Adopt the new messaging infrastructure

### PR DESCRIPTION
**Pull request by ctron**
### Type of change

- Design

### Checklist

- [x] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/design/iot/README.adoc [hunk 1]
```diff
@@ -1,38 +1,148 @@
...
+pointing to exactly one messaging infrastructure.
+* Serves all IoT projects assigned to the IoT infrastructure.
++
+This is an implicit assignment. An IoT project resides in a Kubernetes
+namespace, which is assigned to a messaging tenant, which is assigned
+to a messaging infrastructure.
+
+*Target persona:* IoT service admin
+
+*Proposal:* Change the name from `IoTConfig` to `IoTInfrastructure`
```
Comment (lulf) [inline documentation/design/iot/README.adoc:42]: Agreed on renaming it, just noting we should either call it IoTInfra or rename MessagingInfra to MessagingInfrastructure

File: documentation/design/iot/README.adoc [hunk 2]
```diff
@@ -1,38 +1,148 @@
...
+
+=== Missing messaging tenant
+
+*Aspect:* Getting started
+
+If an IoT project gets created in a namespace which does not have
+a _message tenant_ created, then the IoT project will:
+
+* Get the field `.status.phase` set to `Failed`
+* Provide a helpful message in the field `.status.phaseReason`
```
Comment (lulf) [inline documentation/design/iot/README.adoc:105]: Again pointing out an inconsistency in iot vs enmasse with .status.message vs .status.phaseReason, where we should agree on one of them.

File: documentation/design/iot/README.adoc (hunk 1 above)
Comment (ctron) [reply documentation/design/iot/README.adoc:42]: I would propose then to go for the full name. I don't think it s very common to abbreviate in the Kubernetes world.

File: documentation/design/iot/README.adoc (hunk 2 above)
Comment (ctron) [reply documentation/design/iot/README.adoc:105]: Taking at look at the Kubernetes docs:

> Some resources in the v1 API contain fields called `phase`, and associated `message`, `reason`, and other status fields.

So I think it should be either `reason` or `message`, but not `phaseReason`. If we already have `message` elsewhere, I would propose to switch to `message` here as well.

Comment (ctron) [reply documentation/design/iot/README.adoc]: Done.

File: documentation/design/iot/README.adoc (hunk 1 above)
Comment (kornys) [reply documentation/design/iot/README.adoc:42]: anyway add shortnames please into CRDs :)

**Pull request merged**